src/App.jsx

```diff
@@ -1,5 +1,124 @@
+import { useState, useEffect } from 'react';
+import TodoInput from './components/TodoInput';
+import TodoList from './components/TodoList';
+
+// 아래 @jsxImportSource 이걸 안해주면 css props를 못 받아온다.
+/** @jsxImportSource @emotion/react */
+import { css } from '@emotion/react';
+
 function App() {
-  return <div>React Todo</div>;
+  // 로컬 스토리지에서 저장된 할 일 목록 불러오기
+  const [todos, setTodos] = useState(() => {
+    const savedTodos = localStorage.getItem('todos');
+    return savedTodos ? JSON.parse(savedTodos) : [];
+  });
+
+  // 할 일 목록이 변경될 때마다 로컬 스토리지에 저장
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  const addTodo = (text) => {
+    const newTodo = { id: Date.now(), text, completed: false }; // id에 고유한 숫자를 부여하기 위해서 Date.now() 사용
+    setTodos([...todos, newTodo]);
```

Comment:
`spread` 연산자 사용 아주 좋습니다! 다만 `setState`를 사용할 때 변경되는 state를 직접 참조하여 사용하는 것보다는 콜백 함수를 작성하는 방식이 조금 더 안정적인 방식입니다 🙂

```suggestion
    setTodos((prev) => [...prev, newTodo]);
```

---

_그 이유가 궁금하다면..._

프로젝트를 진행하다 보면 `state 변경 -> 렌더링 -> state 변경 -> 렌더링 -> ...` 과정을 반복하게 됩니다.

`setState(state + 1)`은 현재의 state 값을 기반으로 상태를 업데이트 합니다. 하지만 이 방식은 state 값이 변경되는 과정에서 다른 setState 호출이 있을 경우, 이전 렌더링 시점의 상태를 기반으로 하기 때문에 최신 상태값을 반영하지 못할 수 있습니다.

반면 `setState(prev => prev + 1)`은 콜백 함수를 사용하여 이전 상태(prev)를 기준으로 상태를 업데이트 합니다. 이 방법은 상태 업데이트가 비동기적으로 처리되는 동안에도 항상 최신 상태를 반영하므로, 상태가 여러 번 업데이트 될 때에도 정확한 결과를 보장한다고 합니다!

Reply:
코드 작동 과정에 대한 자세한 설명 너무 감사합니다!😍
비동기적 방식 및 콜백함수를 통해 정확한 결과를 보장하는 코드 작성이 체화되도록 많이 연습해야겠어요 😭
감사합니다👍